modules/generative-palm/clients/palm.go

```diff
@@ -344,7 +344,21 @@ func (v *palm) parseResponse(statusCode int, bodyBytes []byte, debug *modulecapa
 }
 
 func (v *palm) getResponseParams(params map[string]interface{}) map[string]interface{} {
-	return map[string]interface{}{"google": params}
+	return map[string]interface{}{googleparams.Name: params}
+}
+
+func GetResponseParams(result map[string]interface{}) *responseParams {
```

Comment:
does this method need to be public?

Reply:
I have those methods as public so that the gRPC package can extract the correct type from the `map[string]any` that it is working with, which comes from the `traverser.GetClass` method at the root of the `Search` method

My thought was to keep `responseParams` private and have the public method allowing for this requirement. The alternative I saw was to have `ResponseParams` in this package and then `getGoogleResponseParams` in the gRPC package but in my view this would lead to code in different packages depending on the specific format of the `result map[string]any` variable

In this way, the definition of the returned `map[string]any` var, which happens on L347 here can coexist next to the function that maps it back into the `responseParams` type from which it was originally mapped

Hopefully that makes sense 😅